src/main/scala/lms/thirdparty/cuda/cuda.scala

```diff
@@ -177,6 +180,26 @@ object CUDATypeLess extends Dsl with StackArrayOps with CLibs with CudaFunction
     }
   }, m.arrayManifest, m.arrayManifest, m.arrayManifest, manifest[Int])
 
+  def CUDA_TRANSPOSE_KERNEL(m: Manifest[_])(implicit __pos: SourceContext) = CUDA_KERNEL3({xn: List[Backend.Exp] =>
+    withComment(s"generating kernel function for TRANS of type $m") {
+      // type cast
+      val in = (new ARRAY(xn(0))).withSrcType(__pos, m.arrayManifest)
+      val out = (new ARRAY(xn(1))).withSrcType(__pos, m.arrayManifest)
+      val size = (new INT(xn(2))).withSrcType(__pos, manifest[Int])
+
+      val x = blockIdxX * tileDim + threadIdxX
+      val y = blockIdxY * tileDim + threadIdxY
+      val width = gridDimX * tileDim
+
+      val start = Backend.Const(0)
+      for (i <- range_until_step(Wrap[Int](start), Wrap[Int](tileDim.x), Wrap[Int](blockSize.x))) {
+        val index = INT((y + INT(Unwrap(i))) * width + x)
+        out(index) = in(index); ()
```

Comment:
this seems a bit strange.
does this make `out` the same as `in`?

Reply:
oh I accidentally committed the wrong code when I was testing around. I updated the code based on the naive transpose from: https://developer.nvidia.com/blog/efficient-matrix-transpose-cuda-cc/

Reply:
I committed an experiment rather than I was intending on doing here by accident. I have gone ahead and corrected it. My source: https://developer.nvidia.com/blog/efficient-matrix-transpose-cuda-cc/